src/main/java/io/aiven/elasticsearch/gcs/storage/GcsBlobContainer.java

```diff
@@ -0,0 +1,265 @@
+    public DeleteResult delete() throws IOException {
+        final var deletePath = path().buildAsString();
+        LOGGER.info("Delete blob: {}", deletePath);
+        var deleteResult = DeleteResult.ZERO;
+        var page = getBucket().list(Storage.BlobListOption.prefix(deletePath));
+        do {
+            var bytesToDelete = 0L;
+            final var blobsToDeleteBuilder = ImmutableList.<String>builder();
+            for (final var blob : page.getValues()) {
+                bytesToDelete += blob.getSize();
+                blobsToDeleteBuilder.add(blob.getName());
+            }
+            final var blobsToDelete = blobsToDeleteBuilder.build();
+            deleteBlobsIgnoringIfNotExists(blobsToDelete);
+            deleteResult = deleteResult.add(blobsToDelete.size(), bytesToDelete);
+            page = page.getNextPage();
+        } while (Objects.nonNull(page));
+        return deleteResult;
+    }
+
+    private String blobPath(final String blobName) {
+        return path().buildAsString() + blobName;
+    }
+
+    @Override
+    public void deleteBlobsIgnoringIfNotExists(final List<String> blobNames) throws IOException {
+        LOGGER.info("Delete blobs: {}", blobNames);
+        if (blobNames.isEmpty()) {
+            return;
+        }
+
+        final var storageBatch = gcsBlobStore.client().batch();
+        final var storageExceptionHandler = new AtomicReference<StorageException>();
+        blobNames.forEach(name ->
+                storageBatch
+                        .delete(BlobId.of(bucketName, name))
+                        .notify(new BatchResult.Callback<>() {
+                            @Override
+                            public void success(final Boolean result) {
+                            }
+
+                            @Override
+                            public void error(final StorageException exception) {
+                                if (exception.getCode() != HTTP_NOT_FOUND) {
+                                    if (!storageExceptionHandler.compareAndSet(null, exception)) {
+                                        storageExceptionHandler.get().addSuppressed(exception);
+                                    }
+                                }
+                            }
+
+                        })
+        );
+        storageBatch.submit();
+        if (Objects.nonNull(storageExceptionHandler.get())) {
+            throw storageExceptionHandler.get();
+        }
+    }
+
+    @Override
+    public Map<String, BlobMetaData> listBlobs() throws IOException {
+        return listBlobsByPrefix("");
+    }
+
+    @Override
+    public Map<String, BlobMetaData> listBlobsByPrefix(final String blobNamePrefix) throws IOException {
+        final var mapBuilder = ImmutableMap.<String, BlobMetaData>builder();
+        final var prefix = path + blobNamePrefix;
+        LOGGER.info("List of blobs by prefix: {}", prefix);
+        bucketBlobsIterator(prefix)
+                .forEach(blob -> {
```

Comment:
FWIW, the plain `for` is 2 levels of indentation less :) The same below

Reply:
fixed